### PR TITLE
fix: incorrect parsing around '_' character

### PR DESCRIPTION
**Pull request by qexat**

### CODE REVIEW

File: syntaxes/samarium.tmLanguage.json [hunk 1]
```diff
@@ -111,22 +91,42 @@
                     }
                 },
                 {
-                    "match": "([A-Za-z0-9]+|=>)((?:\\s*[A-Za-z0-9]+\\s*(?:\\?|\\.{3})?)*)\\s*(\\*)\\s*",
+                    "match": "([A-Za-z0-9_]+|=>)((?:\\s*[A-Za-z0-9_]+\\s*(?:\\?|\\.{3})?)*)\\s*(\\*)\\s*",
```
Comment (trag1c) [inline syntaxes/samarium.tmLanguage.json]: Just change `[A-Za-z0-9_]` to `\w` at this point

**Review by trag1c — CHANGES_REQUESTED** [review]

File: syntaxes/samarium.tmLanguage.json (hunk 1 above)
Comment (qexat) [reply syntaxes/samarium.tmLanguage.json]: true lmao

Comment (qexat) [reply syntaxes/samarium.tmLanguage.json]: should i do it for classes too?

Comment (trag1c) [reply syntaxes/samarium.tmLanguage.json]: everywhere ig

File: syntaxes/samarium.tmLanguage.json [hunk 2]
```diff
@@ -111,22 +91,42 @@
...
+                    "match": "\\w+"
+                }
+            ]
+        },
+        "operators": {
+            "patterns": [
+                {
+                    "name": "constant.language.samarium",
+                    "match": "_"
+                },
```
Comment (trag1c) [inline syntaxes/samarium.tmLanguage.json]: This is supposed to be removed btw

**Review by trag1c — CHANGES_REQUESTED** [review]

File: syntaxes/samarium.tmLanguage.json (hunk 2 above)
Comment (qexat) [reply syntaxes/samarium.tmLanguage.json]: gg

**Review by trag1c — APPROVED** [review]

**Pull request merged**